test/unit/RAPIDClientTest.js

```diff
@@ -115,3 +132,88 @@ describe('invalid request id works', () => {
     });
   });
 });
+
+describe('next invocation with native client works', () => {
+  it('should call the native client next() method', async () => {
+    const mockNative = new MockNativeClient({
+      bodyJson: '',
+      headers: {
+        'lambda-runtime-aws-request-id': 'test-request-id',
+      },
+    });
+    const client = new RAPIDClient('notUsed:1337', undefined, mockNative);
+    client.useAlternativeClient = false;
+
+    await client.nextInvocation();
+    // verify native client was called
+    mockNative.called.should.be.true();
+  });
+  it('should parse all required headers', async () => {
+    const mockResponse = {
+      bodyJson: '{"message":"Hello from Lambda!"}',
+      headers: {
+        'lambda-runtime-aws-request-id': 'test-request-id',
+        'lambda-runtime-deadline-ms': 1619712000000,
+        'lambda-runtime-trace-id': 'test-trace-id',
+        'lambda-runtime-invoked-function-arn': 'test-function-arn',
+        'lambda-runtime-client-context': '{"client":{"app_title":"MyApp"}}',
+        'lambda-runtime-cognito-identity':
+          '{"identityId":"id123","identityPoolId":"pool123"}',
+        'lambda-runtime-aws-tenant-id': 'test-tenant-id',
+      },
+    };
+
+    const mockNative = new MockNativeClient(mockResponse);
+    const client = new RAPIDClient('notUsed:1337', undefined, mockNative);
+
+    client.useAlternativeClient = false;
+    const response = await client.nextInvocation();
+
+    // Verify all headers are present
+    response.headers.should.have.property(
+      'lambda-runtime-aws-request-id',
+      'test-request-id',
+    );
+    response.headers.should.have.property(
+      'lambda-runtime-deadline-ms',
+      1619712000000,
+    );
+    response.headers.should.have.property(
+      'lambda-runtime-trace-id',
+      'test-trace-id',
+    );
+    response.headers.should.have.property(
+      'lambda-runtime-invoked-function-arn',
+      'test-function-arn',
+    );
+    response.headers.should.have.property(
+      'lambda-runtime-client-context',
+      '{"client":{"app_title":"MyApp"}}',
+    );
+    response.headers.should.have.property(
+      'lambda-runtime-cognito-identity',
+      '{"identityId":"id123","identityPoolId":"pool123"}',
+    );
+    response.headers.should.have.property(
+      'lambda-runtime-aws-tenant-id',
+      'test-tenant-id',
+    );
+    // Verify body is correctly passed through
+    response.bodyJson.should.equal('{"message":"Hello from Lambda!"}');
+  });
+  it('should handle native client errors', async () => {
+    const nativeClient = new MockNativeClient({});
+    nativeClient.shouldThrowError = true;
+
+    const client = new RAPIDClient('localhost:8080', null, nativeClient);
+    client.useAlternativeClient = false;
+
+    try {
+      await client.nextInvocation();
+      // Should not reach here
+      false.should.be.true('Expected an error but none was thrown');
```

Comment:
you can use something like this instead: `await client.nextInvocation().should.be.rejectedWith('Failed to get next invocation');
`